packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -465,6 +470,12 @@ export class EditableGrid extends PureComponent<EditableGridProps, EditableGridS
         document.removeEventListener('paste', this.onPaste);
     }
 
+    componentDidUpdate(prevProps: EditableGridProps): void {
+        if (prevProps.saveBtnClickedCount !== this.props.saveBtnClickedCount) {
```

Comment:
I am confused why this `saveBtnClickedCount` is needed now but wasn't before? There was already the tabbed editable grid with bulk update vs edit individually in this add/move to storage case previously right?

Reply:
This is a workaround to allow buttons (modal-footer) defined outside of EditableGrid, to avoid modal-footer nested under modal-body.

Reply:
But didn't we already have that case for the modal footer outside of the editable grid? I'm just trying to understand why this one was new/different.

Reply:
So prior to this change, the Add to Multiple target editable grid component is relying on EditableGrid.renderButtons() to render the modal buttons, resulting in a wrong DOM with modal-footer inside modal-body. The rendering of buttons inside EditableGrid was implemented due to difficulty with state passing between it and the outer component, as we discussed in a chat with @labkey-alan.

With this PR, I made the change so MultiTargetStorageEditableGrid renders EditableGrid with no tabAdditionalBtn and provides its own footer. But states are still handled in EditableGrid, so I need to notify  EditableGrid that the outer button is clicked with saveBtnClickedCount.

Reply:
I see. Thanks for the info.